fibbingnode/southbound/lsdb.py

```diff
@@ -132,21 +133,29 @@ def endpoints(self, lsdb):
 
 
 class LSAHeader(object):
-    def __init__(self, routerid, linkid, lsa_type, mask, age):
+    def __init__(self, routerid, linkid, lsa_type, mask, age, lsa_seqnum):
         self.routerid = routerid
         self.linkid = linkid
         self.lsa_type = lsa_type
         self.mask = mask
         self.age = int(age)
+        self.lsa_seqnum = self.int_to_2comp(int(lsa_seqnum), 32)
+
+    @staticmethod
+    def int_to_2comp(value, bits):
+        """Compute the 2's compliment of int value val"""
+        if (value & (1 << (bits - 1))) != 0:
+            value = value - (1 << bits)
+        return value
```

Comment:
You can simplify this by directly computing the 2-complement  and returning it (as you compute half of it anyway for the if):
mask = 2 ** (bits - 1)
return -(value & mask) + (value & ~mask)